girder/utility/acl_mixin.py

```diff
@@ -20,32 +20,103 @@
 import itertools
 import six
 
-from ..models.model_base import Model, AccessException
+from ..models.model_base import Model, AccessException, GirderException
 from ..constants import AccessType
 
 
 class AccessControlMixin(object):
     """
-    This mixin is intended to be used for resources which aren't access
-    controlled by default, but resolve their access controls through other
-    resources. As such, the overridden methods retain the same parameters and
-    only alter functionality related to access control resolution.
-
-    resourceColl corresponds to the resource collection that needs to be used
-    for resolution, for example the Item model has a resourceColl of folder.
-
-    resourceParent corresponds to the field in which the parent resource
-    belongs, so for an item it would be the folderId.
+    This mixin is intended to be used for Model types which aren't access controlled by default,
+    but resolve their access controls through another "resource" model object. As such, the
+    overridden methods retain the same parameters and only alter functionality related to access
+    control resolution.
+
+    The "resource" object can be specified in two different ways:
+
+    First:
+      * By setting `resourceColl` on the AccessControlMixin model class, to the "model type
+        identifier" of the resource model.
+      * By setting `resourceParent` on the AccessControlMixin model class, to the name of a field
+        containing the ID of the "resource" object to use for permission checking. This field must
+        be set in every object of the AccessControlMixin model collection.
+    For example:
+    ```
+        class Item(AccessControlMixin, Model):
+            resourceColl = 'folder'
+            resourceParent = 'folderId'
+            ...
+
+        class Folder(AccessControlledModel):
+            ...
+
+        itemInstance = {
+            'folderId': folderInstance['_id'],
+            ...
+        }
+        Item.hasAccess(itemInstance, ...)  # This will check permissions for folderInstance
+    ```
+
+    Second (taking precedence over the first):
```

Comment:
Since this way takes precedence, would it be conceptually better for it to appear first?

Reply:
The first method is actually more common (e.g. used for `File`) and cleaner for downstreams to use, so I wanted to introduce it first.